lib/data/kinetic/Kinetic.js

```diff
@@ -0,0 +1,360 @@
+import crypto from 'crypto';
+import EventEmitter from 'events';
+import net from 'net';
+import stream from 'stream';
+
+import { errors } from 'arsenal';
+import kineticlib from 'kinetic-js';
+
+const HEADER_SZ = 9;
+
+class Kinetic {
+
+    /**
+     * @constructor
+     * @param {Object.<string>[]} drives - the list of drives
+     * @returns {Kinetic} - to allow for a functional style.
+     */
+    constructor(drives) {
+        this._sockets = [];
+        this._drives = drives;
+        this._sockets = this._drives.map(drive => ({
+            drive,
+            clusterVersion: 0,
+            connectionID: 0,
+            sock: this._newSocket(),
+            _request: false,
+            _requestGet: false,
+            _events: new EventEmitter(),
+        }));
+        this._chunkTab = [];
+        this._count = 0;
+        this._lexiKey = '0';
+        this.connect();
+        return this;
```

Comment:
Returning `this` in a constructor is an anti-pattern. It is an internal method meant to return a new instance of the class in any case.